library/core/src/main/java/com/google/android/exoplayer2/text/ttml/TtmlNode.java

```diff
@@ -172,11 +181,37 @@ private void getEventTimes(TreeSet<Long> out, boolean descendsPNode) {
   }
 
   public List<Cue> getCues(long timeUs, Map<String, TtmlStyle> globalStyles,
-      Map<String, TtmlRegion> regionMap) {
+      Map<String, TtmlRegion> regionMap, Map<String, String> imageMap) {
+
     TreeMap<String, SpannableStringBuilder> regionOutputs = new TreeMap<>();
+    List<Pair<String, String>> regionImageList = new ArrayList<>();
+
     traverseForText(timeUs, false, regionId, regionOutputs);
     traverseForStyle(timeUs, globalStyles, regionOutputs);
+    traverseForImage(timeUs, regionId, regionImageList);
+
     List<Cue> cues = new ArrayList<>();
+
+    // Create image based cues
+    for (Pair<String, String> regionImagePair : regionImageList) {
+      String base64 = imageMap.get(regionImagePair.second);
```

Comment:
It's probably worth handling the case that `imageMap.get` returns `null` (i.e. do nothing in this case, because the image ref didn't point to a valid image).